templates/subsample.html

```diff
@@ -58,7 +58,7 @@
 </div>
 
 <hr>
-<h4>Chemical Analyses{% if auth_token %}<span style="float:right"><a href="{{ url_for('edit_chemical_analysis', id='new')}}?subsample_id={{ subsample.id }}">Add Chemical Analysis</a></span>{% endif %}</h4>
+<h4>Chemical Analyses{% if auth_token %}<span style="float:right"><a href="{{ url_for('edit_chemical_analysis', id='new', subsample_id = subsample['id'] )}}?subsample_id={{ subsample.id }}">Add Chemical Analysis</a></span>{% endif %}</h4>
```

Comment:
All of this nesting is intimidating and it seems like you had to debug a lot to identify these things. Would it be worth adding small comments above these calls that explain the arguments and why they were chosen?